projects/ng-live-docs/src/compodoc/compodoc-retriever.service.ts

```diff
@@ -55,11 +55,45 @@ export class CompoDocRetrieverService implements DocumentationRetrieverService {
 
     public getInputParameters(component: Type<unknown>): ApiParameters[] {
         const comp = this.getComponent(component);
-        return comp.inputsClass || [];
+        return comp.inputsClass ? comp.inputsClass.map((input) => {
+            return {...input, typeLink: this.getTypeLink(input.type)};
+        }) : [];
     }
 
     public getOutputParameters(component: Type<unknown>): ApiParameters[] {
         const comp = this.getComponent(component);
-        return comp.outputsClass || [];
+        return comp.outputsClass ? comp.outputsClass.map((output) => {
+            return {...output, typeLink: this.getTypeLink(output.type)};
+        }) : [];
+    }
+
+    private getTypeLink(type: string): string {
+        // Extract the outermost Typescript type of the given type string.
+        const rawType = type.split('[')[0].split('<')[0];
+        // Check every documentation JSON registered with the library.
+        for (const documentationJson of this.documentationJson) {
+            // Check every documentation type in the given documentation
+            for (const typeLink of Object.keys(documentationJson)) {
```

Comment:
I think this is style to visit every type of doc type causes the rest of this code to lose type safety and becomes hard to read. The rest of this class is type safe and we're doing a lot of bracket access here.